### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by MimiUmoh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,34 @@
 # SOM_Weather_Forecast_API
+
+###Description 
+This weather forecast application was created to conduct unit tests on the weather Api.
```
Comment (richgurney) [inline README.md]: These are not unit tests and it is not a application. We have created a SOM to enable us to easily write integration / api tests.

File: README.md [hunk 2]
```diff
@@ -1 +1,34 @@
...
+
+
+##Languages used 
+
+#The language used for this project was Ruby.
+#The gems required for this project were http,json, and rspec.
+
+### How to download
+1. git clone the the repo unto your terminal and then save it or download it as a zip file on your computer.
```
Comment (richgurney) [inline README.md]: This does make sense `unto`. And then save it doesnt make sense. What do you mean?

File: README.md [hunk 3]
```diff
@@ -1 +1,34 @@
...
+
+### How to use it 
+1. To use the weather application, you need to get your weather API key for both the current weather and  the five_day or three hour weather forcasts. You can create your own account and get you own api key.  You can use these links to help you https://openweathermap.org/current),https://openweathermap.org/forecast5.
+2. Once you hvae sorted that out you can then use your API. You can then pick a city ID and city (name) to test.
+3. to Create methods for yout test you go to the services folder in the lib file. You can create methods for both you five day forecast and current forcast there.
```
Comment (richgurney) [inline README.md]: Some of the grammar here needs some attention

File: README.md [hunk 4]
```diff
@@ -1 +1,34 @@
...
+4. You write tests within the spec folder
+5. Once you have done all of this you cna run rspec in the terminal to seeyour reuslts. You will be otld whether or not there were any failures in your test.
```
Comment (richgurney) [inline README.md]: There are  lot of spelling mistakes here

File: lib/weather.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+require_relative 'services/current_weather'
```
Comment (richgurney) [inline lib/weather.rb:1]: These files do not exist

File: spec/current_weather_spec.rb [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+describe Weather do
+  context 'Current weather tetsing  by name' do 
+
+    before(:all) do 
+     @weather = Weather.new.current 
+     @weather.current_weather_city_name('Novinki,ru')
+    end  
+
+    it 'Latitude should be a float ' do
```
Comment (richgurney) [inline spec/current_weather_spec.rb:11]: the format here should be   -----    it 'should return latitude as a float' do

File: spec/current_weather_spec.rb [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    before(:all) do 
+     @weather = Weather.new.current 
+     @weather.current_weather_city_name('Novinki,ru')
+    end  
+
+    it 'Latitude should be a float ' do 
+      expect(@weather.get_latitude).to be_kind_of(Float)
+    end 
+
+    it 'Weather description should be a string' do
```
Comment (richgurney) [inline spec/current_weather_spec.rb:15]: the format here should be ------  it 'should return description as a string'

File: spec/current_weather_spec.rb [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+      expect(@weather.get_weather_description).to be_kind_of(String)
+    end 
+
+    it "icon  in weather should have numbers" do
```
Comment (richgurney) [inline spec/current_weather_spec.rb:19]: Format as above needs fixing ^^^^

File: spec/current_weather_spec.rb [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+      expect(@weather.icon.gsub(/['a'..'z']/,'').to_i).to be_kind_of(Integer)
+    end  
+
+    it "icon  in weather should be a string" do
```
Comment (richgurney) [inline spec/current_weather_spec.rb:23]: Format as above needs fixing ^^^^

File: spec/current_weather_spec.rb [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+      expect(@weather.icon).to be_kind_of(String)
+    end 
+
+    it 'weather id shoud be integer and should equal 701' do
```
Comment (richgurney) [inline spec/current_weather_spec.rb:27]: Format as above needs fixing ^^^^

File: spec/current_weather_spec.rb [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+    it "icon  in weather should be a string" do
+      expect(@weather.icon).to be_kind_of(String)
+    end 
+
+    it 'weather id shoud be integer and should equal 701' do 
+      expect(@weather.get_weather_id).to be_kind_of(Integer)
+    end 
+
+    it ' Id should be an integer and ID should equal 499099' do
```
Comment (richgurney) [inline spec/current_weather_spec.rb:31]: Format as above needs fixing ^^^^

File: spec/current_weather_spec.rb [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+      expect(@weather.get_id).to be_kind_of(Integer).and eq 499099
+    end 
+
+    it 'Temp should be a float' do
```
Comment (richgurney) [inline spec/current_weather_spec.rb:35]: Format as above needs fixing ^^^^

File: spec/current_weather_spec.rb [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+
+    it ' Id should be an integer and ID should equal 499099' do
+      expect(@weather.get_id).to be_kind_of(Integer).and eq 499099
+    end 
+
+    it 'Temp should be a float' do 
+      expect(@weather.get_temp).to be_kind_of(Float)
+    end 
+
+    it 'City name should be Novinki and it should be a string' do
```
Comment (richgurney) [inline spec/current_weather_spec.rb:39]: Format as above needs fixing ^^^^

File: spec/current_weather_spec.rb [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+      expect(@weather.get_city_name).to be_kind_of(String)
+    end 
+
+    it 'type should be an integer' do
```
Comment (richgurney) [inline spec/current_weather_spec.rb:43]: Format as above needs fixing ^^^^

File: spec/current_weather_spec.rb [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+
+    it 'City name should be Novinki and it should be a string' do
+      expect(@weather.get_city_name).to be_kind_of(String)
+    end 
+
+    it 'type should be an integer' do 
+      expect(@weather.get_type).to be_kind_of(Integer)
+    end 
+
+    it 'Country code should be RU' do
```
Comment (richgurney) [inline spec/current_weather_spec.rb:47]: Format as above needs fixing ^^^^

File: spec/current_weather_spec.rb [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+      expect(@weather.get_country_code).to eq "RU"
+    end 
+
+    it 'COD should be 200' do
```
Comment (richgurney) [inline spec/current_weather_spec.rb:51]: What is this refering to? Format as above needs fixing ^^^^

File: spec/current_weather_spec.rb [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+
+    it 'Country code should be RU' do
+      expect(@weather.get_country_code).to eq "RU"
+    end 
+
+    it 'COD should be 200' do 
+      expect(@weather.get_cod).to eq 200
+    end 
+
+    it 'sunrise should have less than 12 characters' do
```
Comment (richgurney) [inline spec/current_weather_spec.rb:55]: Format as above needs fixing ^^^^

File: spec/current_weather_spec.rb [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+    end 
+  end 
+
+  context 'Current Weather testing by ID' do 
+    before(:all) do 
+      @weather_city_id = Weather.new.current 
+      @weather_city_id.current_weather_id('519188')
+     end 
+
+     it 'Latitude should be a float ' do
```
Comment (richgurney) [inline spec/current_weather_spec.rb:66]: Format as above needs fixing ^^^^

File: spec/five_day_weather_spec.rb [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+      @weather.five_day_forecast_by_id('519188')
+      end  
+    
+      it'temperature should be a float' do 
+        expect(@weather.get_temp).to be_kind_of(Float)
+      end 
+
+      it 'temp_kf can be a negative or positive' do
+        expect(@weather.get_temp_kf).to be_kind_of(Float)
+        expect(@weather.get_temp_kf).to be_between(-1000,1000)
```
Comment (richgurney) [inline spec/five_day_weather_spec.rb:17]: Is this a valid temperature range?

**Review by richgurney — CHANGES_REQUESTED** [review]
Hi Emem, there is missing files in the code so when I run it does work as it cannot find the files in services. The test descriptions need a lot of work to be in the correct format. The readme has a lot of spelling and grammar mistakes. Please can you fix there are that i have commented on.

File: lib/weather.rb (hunk 5 above)
Comment (MimiUmoh) [reply lib/weather.rb:1]: Yeah you are right. I am not to sure why it is not working, it works on my laptop. I think some files are missing, I really don't know why that is to be honest.

**Pull request merged**